core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -275,6 +276,25 @@ public static FileIO loadFileIO(
     return fileIO;
   }
 
+  public static MetricsContext loadFileIOMetricsContext(
+      String impl,
+      Map<String, String> properties) {
```

Comment:
this should also have the `Object hadoopConf` in the parameter like `loadFileIO`, we will pass in `null` in S3FileIO but other implementations might leverage this.